homeassistant/components/airvisual/config_flow.py

```diff
@@ -85,33 +95,36 @@ def async_get_options_flow(config_entry):
         """Define the config flow to handle options."""
         return AirVisualOptionsFlowHandler(config_entry)
 
-    async def async_step_geography(self, user_input=None):
+    async def async_step_geography(self, user_input, integration_type):
         """Handle the initialization of the integration via the cloud API."""
-        if not user_input:
-            return self.async_show_form(
-                step_id="geography", data_schema=self.geography_schema
-            )
-
         self._geo_id = async_get_geography_id(user_input)
         await self._async_set_unique_id(self._geo_id)
         self._abort_if_unique_id_configured()
+        return await self.async_step_geography_finish(user_input, integration_type)
 
-        # Find older config entries without unique ID:
-        for entry in self._async_current_entries():
-            if entry.version != 1:
-                continue
+    async def async_step_geography_by_coords(self, user_input=None):
+        """Handle the initialization of the cloud API based on latitude/longitude."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="geography_by_coords", data_schema=self.geography_coords_schema
+            )
 
-            if any(
-                self._geo_id == async_get_geography_id(geography)
-                for geography in entry.data[CONF_GEOGRAPHIES]
-            ):
-                return self.async_abort(reason="already_configured")
+        return await self.async_step_geography(
+            user_input, INTEGRATION_TYPE_GEOGRAPHY_COORDS
+        )
 
-        return await self.async_step_geography_finish(
-            user_input, "geography", self.geography_schema
+    async def async_step_geography_by_name(self, user_input=None):
+        """Handle the initialization of the cloud API based on city/state/country."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="geography_by_name", data_schema=GEOGRAPHY_NAME_SCHEMA
+            )
+
+        return await self.async_step_geography(
+            user_input, INTEGRATION_TYPE_GEOGRAPHY_NAME
         )
 
-    async def async_step_geography_finish(self, user_input, error_step, error_schema):
+    async def async_step_geography_finish(self, user_input, integration_type):
```

Comment:
There's no form for this method, so it doesn't need to be a step. It can just be a helper method. Step methods should only take a single parameter, `user_input`.